onnx/defs/nn/defs.cc

```diff
@@ -3341,6 +3341,8 @@ do not contain padded tokens.
 case, the `K` and `V` inputs correspond to the entire cache tensor, so the four optional inputs/outputs past and
 present key and value should not be used. An additional input `nonpad_kv_seqlen` of shape (batch_size,) may be
 provided to indicate the number of non-padding tokens in each sample of the batch to save unnecessary computation.
+Here, the kv_sequence dimension of `attn_mask` can be shorter than `K` and `V`, but still needs to be at least as long
```

Comment:
If we want to generalize this (how important is it to do this?), then we must also update the documentation of the input, which says it has total-sequence-length. We may need to update the shape-inference method, in case it checks for this.

So, you want to allow smaller, but not bigger?

Reply:
It doesn't seem like shape inference uses this attn_mask input at all. So that part can probably stay as is.
What are you referring to by "generalize"?
The kv_seqlen dimension can be smaller than the K/V inputs's seqlen dimension in case 2 of kv cache. There should not be any reason for it to be bigger than K/V. Maybe I should update the description of this input to explicitly state the dimensions for the two cases of kv cache usage.